### ISSUE TITLE
Feature: Seed database with fake users 

### ISSUE THREAD

**Issue by Uzay-G** [publiclab/plots2#7215]
When testing the local database, it would be very useful to have some fake data to interact with (users, notes).  We can add fake users to the seeds of the database to generate this fake test content.

I propose we use the [faker gem](https://github.com/faker-ruby/faker) to accomplish this. What do you think @publiclab/community-reps?

**Comment by SidharthBansal** [comment]
You will get 1 point for suggestion for this issue opening.

**Comment by SidharthBansal** [comment]
Similarly goes for other issues.

**Comment by SidharthBansal** [comment]
@Uzay-G what is the need of faker gem?

**Comment by Uzay-G** [comment]
Basically, it generates fake data in the development database to use when working on new features or bugfixing.
For example, the following snippet, if it was placed in the [databases files](https://github.com/publiclab/plots2/blob/master/db/seeds.rb) would generate several fake users:
```ruby
99.times do |n|
  name  = Faker::Name.name
  email = Faker::Internet.email
  password = "password"
  User.create!(username:  name,
               email: email,
               password:              password,
               password_confirmation: password,
               status:  1)
end
```

I would have to change some minor details but this would work :+1:

**Comment by Uzay-G** [comment]
I could also implement this for notes or other posts. 
There are many cases where this is useful. For example, I once worked on an issue about how different notes were being sorted based on likes. To test this feature, I had to manually create several notes to check they were being sorted correctly

**Comment by SidharthBansal** [comment]
Ask jeff.
I don't think we need this although it is a useful gem

**Comment by nstjean** [comment]
I think this would be useful for notes and wikis, to have pre-populated data for developers to work with. When I was working on page tabbing I ended up creating a lot of posts by hand.

**Comment by jywarren** [comment]
I like this; want to note that we do this without the Faker gem already for some tests. Like tags. If Faker is REALLY helpful let's do it, but if we can get by without, let's aim to keep our gem list relatively short, as it adds complexity to the installation over the long run.

**Comment by jywarren** [comment]
I do think this is a good idea!

**Comment by Uzay-G** [comment]
Oh yeah no problem. I think we could do it without the gem with something like this:
```ruby
99.times do |n|
  name  = "user#{n}"
  email = "user#{n}@publiclab.org"
  password = "password"
  User.create!(username:  name,
               email: email,
               password:              password,
               password_confirmation: password,
               status:  1)
end
```

This would make every username and email different. What do you think?

**Comment by emilyashley** [comment]
Is this seeding to be used for local developer environments or for generating data for automated testing? Or both?

Still learning how this system works but I think it'd be cool if it tackles both of those things well. 👍

**Comment by Uzay-G** [comment]
In this case it would only be for the local dev environments but we can use similar methods for testing :+1:

**Comment by emilyashley** [comment]
Oh also when I saw this I remember doing the same thing on other sites!

> I think this would be useful for notes and wikis, to have pre-populated data for developers to work with. When I was working on page tabbing I ended up creating a lot of posts by hand.

I don't know if this is helpful here, but what do instead is modify the pagination settings to 3 items per page or something, then I can test a lot of the pagination features without creating content by hand. I just gotta remember to set it back to 100 or whatever the default is when I commit my code.

**Comment by SidharthBansal** [comment]
Correct Emily, this is the first use case of faker in every rails guide.
We you guys are planning for a pr, complexity wise it should be medium but
there will be a lot of loops for different fixtures. So lines of code will
be higher. So let's keep it hard or five point task for GCI

**Comment by Uzay-G** [comment]
Alright, I will do it. Should I use faker or do we not need it?

**Comment by Uzay-G** [comment]
I will do it once I am done with testing. :+1:

**Comment by nstjean** [comment]
> Oh also when I saw this I remember doing the same thing on other sites! @nstjean Is page tabbing the same thing as pagination?
> 
> > I think this would be useful for notes and wikis, to have pre-populated data for developers to work with. When I was working on page tabbing I ended up creating a lot of posts by hand.
> 
> I don't know if this is helpful here, but what I've done instead when I don't have enough seeds is to modify the pagination settings to 3 items per page or something, then I can test a lot of the pagination features without creating content by hand. I just gotta remember to set it back to 100 or whatever the default is when I commit my code.

Yes sorry, pagination! I believe that's what I ended up doing at the end, and then I wondered why I hadn't thought of that to start with. :laughing:

**Comment by SidharthBansal** [comment]
@Uzay-G are you willing to do it?

**Comment by Uzay-G** [comment]
Yes I do it :smile:

**Comment by SidharthBansal** [comment]
Thanks

**Comment by Uzay-G** [comment]
I will do it without faker because @jywarren said, we want to keep installation simple.

**Comment by Uzay-G** [comment]
I will do it without faker because @jywarren said, we want to keep installation simple.

**Comment by Uzay-G** [comment]
I will do it without faker because @jywarren said, we want to keep installation simple.

**Comment by SidharthBansal** [comment]
So we need to close this?
I am also in favour of closing it as said earlier

**Comment by Uzay-G** [comment]
No I am going to add seeds without faker

**Comment by SidharthBansal** [comment]
ok cool
